src/markup.c

```diff
@@ -44,6 +46,189 @@ static char *markup_br2nl(char *str)
+
+                // the tag is broken, ignore it
+                if (!end) {
+                        fprintf(stderr, "WARNING: Given image is broken: '%s'\n", start);
+                        string_replace_at(*str, start-*str, strlen(start), "");
+                        break;
+                }
+
+                // use attribute=" as stated in the notification spec
+                const char *alt_s = strstr(start, "alt=\"");
+                const char *src_s = strstr(start, "src=\"");
+
+                char *text_alt = NULL;
+                char *text_src = NULL;
+
+                const char *src_e = NULL, *alt_e = NULL;
+                if (alt_s)
+                        alt_e = strstr(alt_s + strlen("alt=\""), "\"");
+                if (src_s)
+                        src_e = strstr(src_s + strlen("src=\""), "\"");
+
+                // Move pointer to the actual start
+                alt_s = alt_s ? alt_s+5 : NULL;
+                src_s = src_s ? src_s+5 : NULL;
+
+                /* check if alt and src attribute are given
+                 * If both given, check the alignment of all pointers */
+                if (   alt_s && alt_e
+                    && src_s && src_e
+                    && (  (alt_s < src_s && alt_e < src_s-5 && src_e < end)
+                        ||(src_s < alt_s && src_e < alt_s-5 && alt_e < end)) ) {
+
+                        text_alt = g_strndup(alt_s, alt_e-alt_s);
+                        text_src = g_strndup(src_s, src_e-src_s);
+
+                /* check if single valid alt attribute is available */
+                } else if (alt_s && alt_e && alt_e < end && (!src_s || src_s < alt_s || alt_e < src_s-5)) {
+                        text_alt = g_strndup(alt_s, alt_e-alt_s);
+
+                /* check if single valid src attribute is available */
+                } else if (src_s && src_e && src_e < end && (!alt_s || alt_s < src_s || src_e < alt_s-5)) {
+                        text_src = g_strndup(src_s, src_e-src_s);
+
+                } else {
+                        fprintf(stderr,
+                                "WARNING: Given image argument is broken: '%.*s'\n",
+                                (int)(end-start), start);
+                }
+
+                // replacement text for alt
+                int repl_len = end - start + 1;
+
+                if (!text_alt)
+                        text_alt = g_strdup("");
+
+                *str = string_replace_at(*str, start-*str, repl_len, text_alt);
+
+                // if there had been a href attribute,
+                // add it to the URLs
+                if (text_src && urls) {
+
+                        // If text_alt is not given, the image is
+                        // invisible in the replaced text
+                        if (!text_alt || strlen(text_alt) == 0) {
+                                g_free(text_alt);
+                                text_alt = g_strdup("Invisible Image");
```

Comment:
Instead of using `Invisible Image` for everything let's fall back to our previous system, numbers.

`Notification with inline image [image #1]`
And in dmenu
`[#1] url`

Reply:
I agree. Setting the text to "Invisible image" sounds rather mysterious than helpful.

Reply:
Sorry, I've seen this just now.

> Notification with inline image [image #1]
> And in dmenu
> [#1] url

Please, no. As a user, I'd question the rationale removing it for links and then adding it again for images. As a user, I'd really insult this dev (myself).

Simply changing it to `[Hidden Image]` makes more sense IMO.